src/io/fasta.rs

```diff
@@ -3,14 +3,101 @@
 // This file may not be copied, modified, or distributed
 // except according to those terms.
 
-//! FASTA format reading and writing.
+//! Structs and trait to read and write file in FASTA format.
 //!
 //! # Example
 //!
+//! ## Read
+//!
+//! In this example, we parse a fasta file from stdin and compute some statistics
+//!
+//! ```
+//! // import functions (at top of script)
+//! use std::io;
+//! use bio::io::fasta;
+//! // call within functions
+//! let mut reader = fasta::Reader::new(io::stdin());
+//!
+//! let mut nb_reads = 0;
+//! let mut nb_bases = 0;
+//!
+//! for result in reader.records() {
+//!     let record = result.expect("Error during fasta record parsing");
```

Comment:
It is possible to use the `?` operator for error handling, but you'd have to define the `Result` type then in a comment and return `Ok(())` at the end of the example. I find myself unable to find an example for that, but it exists... somewhere...

Reply:
I've seen examples like this too. I think this is now the recommended way to handle errors?

```
/// Read JSON from a chunk file
fn read_file<P: AsRef<Path>>(filename: P) -> Result<String, Error> {
    let mut f = File::open(filename.as_ref())?;
    let mut buf = String::new();
    f.read_to_string(&mut buf)?;
    Ok(buf)
}
```

Although since clippy takes issue with `panic` and Rust-lang [doesn't recommend error handling in examples](https://doc.rust-lang.org/rustdoc/documentation-tests.html#using--in-doc-tests), I'm having second thoughts about this.

Reply:
We could change it to `let record = result?;` but we have a lot of `.expect()`. calls in the io examples so let's decide whether to do to that before spending time to change them all.

e.g., 

```
//! for result in reader.records() {
//!     let record = result?;
//!     println!("{}", record.id());
//!
//!     nb_reads += 1;
//!     nb_bases += record.seq().len();
//!
//!     Ok(())
//! }
```

Reply:
I think
```
//! ```
//! use std::io;
//! use bio::io::fasta;
//! let mut reader = fasta::Reader::new(io::stdin());
//!
//! let mut nb_reads = 0;
//! let mut nb_bases = 0;
//!
//! for result in reader.records() {
//!     let record = result?;
//!     println!("{}", record.id());
//!
//!     nb_reads += 1;
//!     nb_bases += record.seq().len();
//! }
//!
//! println!("Number of reads: {}", nb_reads);
//! println!("Number of bases: {}", nb_bases);
//! # Ok::<(), WhateverTheErrorTypeIs>(())
//! ```
```
should be sufficient.

But generally speaking, yes, either we use the `?` operator everywhere (when appropriate) or stick to `expect`.

Reply:
I think it's worth considering migrating to the `?` as it seems to be recommended now. However, other examples in the repo seem to have `.expect()` calls as well (except [here](https://github.com/rust-bio/rust-bio/blob/master/src/io/bed.rs#L245) and [here](https://github.com/rust-bio/rust-bio/blob/master/src/io/bed.rs#L326) in parsing BED files). I think it's better to be consistent now: leave it as `.expect` and discuss whether to migrate over as a separate PR.